design-proposals/golden-image-delivery-and-update-pipeline.md

```diff
@@ -0,0 +1,207 @@
+
+A **VolumeReference** object references an import/clone source for a PVC. Functionally, it is similar to the “source” field of a DataVolume. The VolumeReference’s source is mutable, which has the side effect of allowing the VolumeReference to act as a static link that will redirect to whatever source the link currently points to. In the future, the VolumeReference will act as a PVC Populator as well as a DataVolume source.
+
+**VolumeImportPollers can manage a corresponding VolumeReference** to always point to the latest most up-to-date PVC that the poller has created. 
+
+
+## CDI: DataVolume VolumeReference Source Field
+
+The DataVolume object will add a new source field capable of importing pvc content using a VolumeReference. 
+
+![DataVolume VolumeReference Source Field](golden-image-delivery-images/volume-reference-source.png "DataVolume VolumeReference Source Field")
+
+## SSP: Setup VolumeImportPollers for common templates
+
+Since the SSP operator manages installing common templates, this operator should also be responsible for setting up the VolumeImportPollers used to automatically source the disk images into the cluster that the common templates use.
+
+These VolumeImportPollers can be configured on the SSP operators Custom Resource.
+
+
+![SSP Disk Import Pipelines](golden-image-delivery-images/ssp-disk-import-pipelines.png "SSP Disk Import Pipelines")
+
+
+## Common templates: VolumeReferences as DataVolume Source
+
+The common templates shipped with SSP need to have their DataVolumeTemplate sections transition to using a VolumeReference as the source field rather than a PVC claim name. This allows the VolumeImportPoller to seamlessly update the underlying PVC the corresponding VolumeReference points to, which then ensures that the common templates are always using the most up-to-date disk images.
+
+## HCO: Integration with SSP Operator
+
+The HCO is responsible for setting VolumeImportPoller sources on the SSP operator’s CR. These volume sources are the locations used by the VolumeImportPollers to populate PVCs for the common templates.
+
+
+## OCP UI: Integration with VolumeReferences
+
+The UI currently detects if a template has a boot source by checking if the DataVolumeTemplate’s source PVC is present. The UI needs to now be capable of checking if a DataVolumeTemplate’s source VolumeReference has phase: linked as well in order to determine if a boot source is set.
+
+## Scalability
+
+The only scaling concern is that the automated process has the potential to exhaust storage resources unless a PVC garbage collection mechanism is observed by the VolumeImportPoller.
```

Comment:
High amounts of manually created VolumeImportPollers can also cause network and cpu load at the import intervals.
Especially for clusters that enable or upgrade to this feature it could cause an initial spike in utilization until every image has been fetched once.